components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -1216,11 +1216,15 @@ public Response apisChangeLifecyclePost(String action, String apiId, String life
     @Override
     public Response apisCopyApiPost(String newVersion, String apiId, Request request) throws NotFoundException {
         APIDTO newVersionedApi;
+        String apiName, newApiVersion;
         String username = RestApiUtil.getLoggedInUsername(request);
         try {
             APIPublisher apiPublisher = RestAPIPublisherUtil.getApiPublisher(username);
             String newAPIVersionId = apiPublisher.createNewAPIVersion(apiId, newVersion);
             newVersionedApi = MappingUtil.toAPIDto(apiPublisher.getAPIbyUUID(newAPIVersionId));
+            apiName = newVersionedApi.getName();
+            newApiVersion = newVersionedApi.getVersion();
+            apiPublisher.sendMailNotification(apiPublisher, apiId, newApiVersion, apiName);
```

Comment:
Sending emails is a part of core implementation. Basic idea is that it should run inside apiPublisher.createNewAPIVersion(apiId, newVersion) method. Let's move this to core.